src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -84,7 +83,9 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
     }
     private preferredControllers = new Map<NotebookDocument, VSCodeNotebookController>();
 
-    private readonly isLocalLaunch: boolean;
+    private get isLocalLaunch(): boolean {
+        return isLocalLaunch(this.configuration);
```

Comment:
Will need a better flag, 
If `isLocalLaunch` === true, then tshi means we have a remote uri,
however in this PR, we support local & remote together, ie. even when we're laucning local kernels we can be working against remote kernels.